fix

dianaeldemerdash opened the pull request.

dianaeldemerdash commented:
nosetests3 -vs --logging-level=WARNING ui_testing/testcases/basic_tests/test06_orders.py:OrdersTestCases.test008_duplicate_main_order --tc-file=config.ini --tc=site.password:admin 
2020-06-08 19:30:51.569 | INFO     | api_testing.apis.base_api:info:76 - Get authorized api session.
2020-06-08 19:30:52.624 | INFO     | api_testing.apis.base_api:info:76 - session ID : eyJhbGciOi .....
New: Orders with test units: Duplicate an order with test unit 1 copy ...       
2020-06-08 19:30:52.843 | INFO     | ui_testing.testcases.base_test:info:129 - Test case : test008_duplicate_main_order
2020-06-08 19:31:22.262 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:531 - wait until page is loaded
2020-06-08 19:31:25.596 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/testUnits
2020-06-08 19:31:26.435 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-06-08 19:31:26.436 | INFO     | ui_testing.pages.order_page:create_new_order:114 -  Create new order.
2020-06-08 19:31:26.436 | INFO     | ui_testing.pages.orders_page:click_create_order_button:21 - Press create order button
2020-06-08 19:31:26.997 | INFO     | ui_testing.pages.base_pages:sleep_small:43 -  Small sleep.
2020-06-08 19:31:31.999 | INFO     | ui_testing.pages.order_page:set_new_order:18 - Set new order.
2020-06-08 19:31:45.112 | INFO     | ui_testing.pages.base_pages:sleep_small:43 -  Small sleep.
2020-06-08 19:32:05.429 | INFO     | ui_testing.pages.base_pages:sleep_small:43 -  Small sleep.
2020-06-08 19:32:22.602 | INFO     | ui_testing.pages.base_pages:sleep_small:43 -  Small sleep.
2020-06-08 19:32:35.263 | INFO     | ui_testing.pages.base_pages:info:274 -  save the changes
2020-06-08 19:32:35.310 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-06-08 19:32:37.523 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-06-08 19:32:39.525 | INFO     | ui_testing.pages.order_page:create_new_order:135 -  Order created with no : 9'075'244'469 
2020-06-08 19:32:39.622 | INFO     | ui_testing.pages.order_page:get_suborder_data:326 - getting main order data
2020-06-08 19:32:39.796 | INFO     | ui_testing.pages.order_page:get_suborder_data:333 - getting suborders data
2020-06-08 19:32:44.460 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:531 - wait until page is loaded
2020-06-08 19:32:51.639 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-06-08 19:32:53.810 | INFO     | ui_testing.pages.base_pages:info:274 -  duplicate the main order
2020-06-08 19:32:54.031 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-06-08 19:32:56.791 | INFO     | ui_testing.pages.base_pages:sleep_medium:47 -  Medium sleep.
2020-06-08 19:33:06.944 | INFO     | ui_testing.pages.order_page:get_suborder_data:326 - getting main order data
2020-06-08 19:33:07.094 | INFO     | ui_testing.pages.order_page:get_suborder_data:333 - getting suborders data
2020-06-08 19:33:07.815 | INFO     | ui_testing.pages.base_pages:info:274 -  save the changes
2020-06-08 19:33:07.857 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-06-08 19:33:10.053 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-06-08 19:33:16.034 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:531 - wait until page is loaded
2020-06-08 19:33:28.083 | INFO     | ui_testing.testcases.base_test:info:129 - TearDown.        
ok

----------------------------------------------------------------------
Ran 1 test in 155.241s

OK

0xIslamTaha commented on ui_testing/testcases/basic_tests/test06_orders.py:
```diff
@@ -304,52 +304,48 @@ def test007_order_search(self, small_letters):
+        random_testunit, payload = self.test_unit_api.get_all_test_units(limit=20)
+        test_unit = random.choice(random_testunit['testUnits'])
+
+        data_before_duplicate = self.order_page.create_new_order(contact='', material_type='r', article='a',
+                                                                 test_plans=[],
+                                                                 test_units=[test_unit['name']])
+
+        self.orders_page.get_orders_page()
+        order_no = data_before_duplicate['orderNo']
```
- Use API

dianaeldemerdash replied:
Done

0xIslamTaha commented:
@dianaeldemerdash I'm wondering why did u ignore those test cases?

- https://github.com/modeso-open-source/1lims-automation/pull/232/files#diff-05c34fcc59d5beb73a7b86f1cf41a0c1R410
- https://github.com/modeso-open-source/1lims-automation/pull/232/files#diff-05c34fcc59d5beb73a7b86f1cf41a0c1R857

dianaeldemerdash commented:
nosetests3 -vs --logging-level=WARNING ui_testing/testcases/basic_tests/test06_orders.py:OrdersTestCases.test008_duplicate_main_order --tc-file=config.ini --tc=site.password:admin 
2020-06-10 05:31:01.650 | INFO     | api_testing.apis.base_api:info:76 - Get authorized api session.
2020-06-10 05:31:02.634 | INFO     | api_testing.apis.base_api:info:76 - session ID : eyJhbGciOi .....
New: Orders with test units: Duplicate an order with test unit 1 copy ...       
2020-06-10 05:31:02.827 | INFO     | ui_testing.testcases.base_test:info:129 - Test case : test008_duplicate_main_order
2020-06-10 05:31:39.374 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:531 - wait until page is loaded
2020-06-10 05:31:40.114 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/orders/auto/generatedId
2020-06-10 05:31:40.829 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-06-10 05:31:40.830 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/testPlans
2020-06-10 05:31:41.383 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-06-10 05:31:41.384 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/materialTypes
2020-06-10 05:31:41.739 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-06-10 05:31:41.739 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/articles
2020-06-10 05:31:42.129 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-06-10 05:31:42.129 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/testUnits/get/names/1?name=&negelectIsDeleted=0&searchableValue=
2020-06-10 05:31:42.554 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-06-10 05:31:42.554 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/contacts
2020-06-10 05:31:43.983 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-06-10 05:31:43.983 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/orders
2020-06-10 05:31:44.637 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-06-10 05:31:50.976 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-06-10 05:31:53.174 | INFO     | ui_testing.pages.base_pages:info:274 -  duplicate the main order
2020-06-10 05:31:53.414 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-06-10 05:31:56.138 | INFO     | ui_testing.pages.base_pages:sleep_medium:47 -  Medium sleep.
2020-06-10 05:32:06.294 | INFO     | ui_testing.pages.order_page:get_suborder_data:326 - getting main order data
2020-06-10 05:32:06.455 | INFO     | ui_testing.pages.order_page:get_suborder_data:333 - getting suborders data
2020-06-10 05:32:07.155 | INFO     | ui_testing.pages.base_pages:info:274 -  save the changes
2020-06-10 05:32:07.198 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-06-10 05:32:09.361 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-06-10 05:32:15.158 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:531 - wait until page is loaded
2020-06-10 05:32:27.333 | INFO     | ui_testing.testcases.base_test:info:129 - TearDown.        
ok

----------------------------------------------------------------------
Ran 1 test in 84.507s

OK

dianaeldemerdash commented:
@0xIslamTaha you required from me in the old PR to run all those cases because I changed in a lot of files and for this reason I created new PR and I changed in two files only that belongs to my test case 008_duplicate_main_order so why I need to run the others ? those not from my test case scope
